build vaadin 14.14 project with JDK 21 failed.

### Description of the bug

build project using Vaadin 14.14.0 failed with following error. 

```
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:14.14.0:build-frontend (default) on project docs: Execution default of goal com.vaadin:vaadin-maven-plugin:14.14.0:build-frontend failed: Unsupported class file major version 65 -> [Help 1]

```

### Expected behavior

build passes

### Minimal reproducible example

- git clone git@github.com:vaadin/docs.git
- cd docs
- git checkout v14
- mvn install -DskipTests -Pproduction


### Versions

- Vaadin / Flow version: 14.14
- Java version: 21

--- a/pom.xml
+++ b/pom.xml
@@ -223,7 +223,7 @@
             <dependency>
                 <groupId>net.bytebuddy</groupId>
                 <artifactId>byte-buddy</artifactId>
-                <version>1.17.8</version>
+                <version>1.14.8</version>
             </dependency>
 
             <dependency>

